examples/darwin-framework-tool/commands/provider/OTAProviderDelegate.mm

```diff
@@ -0,0 +1,173 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "OTAProviderDelegate.h"
+#import <Matter/Matter.h>
+
+constexpr uint8_t kUpdateTokenLen = 32;
+
+@interface OTAProviderDelegate ()
+@property NSString * mOTAFilePath;
+@property DeviceSoftwareVersionModel * candidate;
+@end
+
+@implementation OTAProviderDelegate
+
+- (instancetype)init
+{
+    if (self = [super init]) {
+        _nodeID = @(0);
+        _selectedCandidate = [[DeviceSoftwareVersionModel alloc] init];
+        _userConsentState = OTAProviderUserUnknown;
+    }
+    return self;
+}
+
+- (void)handleQueryImage:(MTROtaSoftwareUpdateProviderClusterQueryImageParams * _Nonnull)params
```

Comment:
I don't think I see any checks for the protocols supported?

Reply:
How do I determine which protocol are supported?

Reply:
I created an https://github.com/project-chip/connectedhomeip/issues/20873 for an enhancement request to allow the user to set the supported protocols.